### DISCUSSION TITLE
debug.cr shards not work with lucky docker-compose up, any idea?

### DISCUSSION THREAD

**Discussion by zw963** [luckyframework/lucky#1804] [Feature requests & Ideas] [answered]
We always use our lovely clover project for example.

1. Add debug shards
2. require debug in src/shards.cr
3.  Add following debug code into src/actions/home/index.cr

```cr
class Home::Index < BrowserAction
  include Auth::AllowGuests

  get "/" do
    if current_user?
      redirect Me::Show
    else
      p "1"*30
      debug!("3"*30)
      html Home::IndexPage
    end
  end
end
```
4. start app use `lucky dev` on local laptop directly, we can saw it works as expected.

![image](https://user-images.githubusercontent.com/549126/233832976-60a9076d-9bb8-4127-b295-4de948d819db.png)

 5. start app use docker-compose, the output of debug shards missing, only p output correct.

```
clover-lucky-1     | web          | 2023-04-23T09:14:29.002753Z  DEBUG - lucky:  -- local: {"handled_by" => "Home::Index"}
clover-lucky-1     | web          | "111111111111111111111111111111"
clover-lucky-1     | web          | 2023-04-23T09:14:29.006859Z  DEBUG - lucky: Rendered Home::IndexPage
clover-lucky-1     | web          | 2023-04-23T09:14:29.006990Z   INFO - lucky:  -- local: {"status" => 200, "duration" => "4.25ms", "request_id" => nil}
```

Any idea? thanks

**Comment by Sija** [comment]
Perhaps `DEBUG=1` env variable is not being passed into the docker container?

**Reply by zw963** [reply]
> Perhaps `DEBUG=1` env variable is not being passed into the docker container?

In fact, it has, please check following screenshot.

![image](https://github.com/luckyframework/lucky/assets/549126/ab297a57-7d6c-4f5d-9b8e-3cb88c88bf58)

This sample project available at https://github.com/zw963/clover/blob/master/src/actions/home/index.cr#L16-L18

Thanks

**Reply by Sija** [reply, marked as answer]
This same env needs to be also passed during compilation, is that happening?

**Reply by zw963** [reply]
> This same env needs to be also passed during compilation, is that happening?

Add ENV `DEBUG=1` into docker-compose.yml works for me, thank you very much!
